ndcube/ndcube.py

```diff
@@ -33,6 +33,37 @@ class NDCubeABC(astropy.nddata.NDData, metaclass=NDCubeMetaClass):
     def dimensions(self):
         pass
 
+    @abc.abstractproperty
+    def crop(self, lower_corner, upper_corner, coords_set='wcs', crop_by_values=False):
```

Comment:
I am not convinced by either `coords` or `coords_set` for this kwarg name, but I haven't got a better option right now (other than `wcs`).

Reply:
That's what I started with.  But then I thought `wcs='wcs'` and `wcs='extra coords'` would seem a bit confusing to people who aren't us.